db/migration/1755013960275-AddPostsGdocsLinksTargetIndexes.ts

```diff
@@ -0,0 +1,33 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class AddPostsGdocsLinksTargetIndexes1755013960275
+    implements MigrationInterface
+{
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        // Add index on target column (with prefix length for TEXT column) for improved JOIN performance
+        await queryRunner.query(`
+            CREATE INDEX idx_posts_gdocs_links_target 
+            ON posts_gdocs_links(target(100))
+        `)
```

Comment:
Personally, I would only add the second (composite) index below, and not add this one here.

The composite index can always also be used for queries that benefit from an index on `target(100)`.